derivative/differentiation.py

```diff
@@ -228,6 +228,8 @@ def x(self, X, t, axis=1):
 
 
 def _align_axes(X, t, axis) -> tuple[NDArray, tuple[int, ...]]:
+    """Reshapes the data so the derivative always happens along axis 1. Do we need this?
```

Comment:
Do we need this function? All this reshaping is a bit confusing. Can we just take the data as is and differentiate along whichever axis the user specifies? There are ways of indexing in `numpy` that make this fairly easy usually. For instance I have [this line](https://github.com/pavelkomarov/spectral-derivatives/blob/main/specderiv/specderiv.py#L45).

The code could probably be both more general and simplified if we do this right, but it's a separate PR for sure.

Reply:
Happy for this, but the challenge is that all methods internally have been allowed to expect one particular axis.  So it probably means changing a fair amount.